lib/r10k/api.rb

```diff
@@ -186,22 +186,68 @@ def module_source_for_module(module_name, env_map)
     def module_sources_for_environment(env_map)
     end
 
+    # Update local caches represented by the given by module_sources, a collection of module_source hashmaps.
+    #
+    # @param module_sources [Array<Hash>] An array of hashmaps each representing a single remote module source (as produced by {#module_sources_for_environment})
+    # @param cache_basedir [String] Root of where r10k is caching things.
+    # @return [true] Returns true on success, raises on failure.
+    # @raise [RuntimeError] Something bad happened!
+    def update_caches(module_sources, cache_basedir, opts = {})
+      if module_sources.respond_to?(:each)
+        module_sources.each do |module_source|
+          update_cache(module_source, cache_basedir, opts)
+        end
+        return true
+      else
+        raise RuntimeError.new("module_sources must be a collection of source hashes.")
+      end
+
+    end
     # Update local cache represented by the given module_source hashmap.
     #
     # @param module_source [Hash] A hashmap representing a single remote module source (as produced by {#module_sources_for_environment})
+    # @param cache_basedir [String] Root of where r10k is caching things.
     # @return [true] Returns true on success, raises on failure.
-    # @raise [RuntimeError] Something bad happened!
-    def update_cache(module_source)
+    # @raise [UnresolvableError] Something bad happened!
+    def update_cache(module_source, cache_basedir, opts = {})
+      type = module_source[:type]
+      source = module_source[:source]
+
+      if type
+        result = case type.to_sym
+        when :git
+          opts[:cache_basedir] = cache_basedir
+          update_git_cache(source, opts)
+        when :forge
+          raise NotImplementedError
+        when :svn
+          raise NotImplementedError
+        else
+          raise UnresolvableError.new("Unrecognized module source type '#{module_source[:type]}'.")
+        end
+      end
+      result
     end
 
-    # Update local cache of the given remote VCS repository.
+    # Update local cache of the given remote git repository.
     #
     # @param remote [String] URI for the remote repository which should be cached or updated.
     # @param opts [Hash] Additional options as defined.
     # @option opts [String] :cachedir Base path where caches are stored.
     # @return [true] Returns true on success, raises on failure.
     # @raise [RuntimeError] Something bad happened!
-    def update_vcs_cache(remote, opts={})
+    def update_git_cache(remote, opts={})
```

Comment:
does this function make sense without a cache_basedir? It seems like that should just be an argument.